### ISSUE TITLE
Amazon S3 sender throws IOException: Attempted read on closed stream when action=read but does not when action=readDelete

### ISSUE THREAD

**Issue by Laurens-makel** [ibissource/iaf#4621]
**❗ please do not add sensitive information in issues, you can provide extra information via email using issue number as reference ❗**

**Describe the issue**
When using the following S3 sender configuration (action=read)

```
<SenderPipe name="Read file from external storage" storeResultInSessionKey="file">
        <AmazonS3Sender action="read" bucketName="${aws.s3.bucket.name}"
				        accessKey="${aws.accessKey}" secretKey="${aws.secretKey}" authAlias="${aws.authAlias}">
	        <Param name="filename" sessionKey="DocumentLocation"/>
        </AmazonS3Sender>
        
        <Forward name="success" path="Dummy xslt" />
</SenderPipe>
```

It throws the following error:
![image](https://user-images.githubusercontent.com/37303445/224309853-33f88044-c290-44e9-a14e-646ea98f3307.png)

When switching to action=readDelete, this does not occur and the message is processed as expected.

**Reporter**
Laurens

**Environment**
```
FF! 7.7.3: Ibis4XX
Running on X using Apache Tomcat/9.0.72
Java Version: Java(TM) SE Runtime Environment (1.8.0_341-b10)
Heap size: 881M, total JVM memory: 1448M
Free disk space: 269GB, total disk space: 476GB
Up since: 2023-03-10 12:27:44 (28m)
```

**Comment by Laurens-makel** [comment]
I have tested the above S3 sender as a single pipe in a pipeline, the error occurs when Ladybug is turned on and does not occur when Ladybug is turned off.

When testing in a pipeline with multiple pipes, when Ladybug is turned on, the error occurs right after the pipe execution, when Ladybug is turned off, the error occurs in the pipe that tries to consume the stream.  

![image](https://user-images.githubusercontent.com/37303445/224324354-311833ba-0705-459d-b380-4cf10bce36a0.png)

**Comment by Laurens-makel** [comment]
The following change seems to solve the issue locally:

https://github.com/ibissource/iaf/blob/master/core/src/main/java/nl/nn/adapterframework/filesystem/FileSystemActor.java 
![image](https://user-images.githubusercontent.com/37303445/224334241-b093c301-3678-40af-a738-883266c6ae71.png)

**Comment by jacodg** [comment]
Can you test it with 7.7.6 and/or 7.8?

**Comment by Laurens-makel** [comment]
7.7.6:
![image](https://user-images.githubusercontent.com/37303445/224742883-8760a0e7-93f8-4100-8e75-098220c37d5b.png)

It takes too much time and effort to start upgrading to 7.8 locally only to test this issue.

**Comment by nielsm5** [comment]
Should have been fixed in #4474 though it's a bit more difficult to test. I've added a few tests to check this behavior. The message#preserve method should counter this.
